### DISCUSSION TITLE
Filters for fleets and stars

### DISCUSSION THREAD

**Discussion by anicolao** [Tsangares/stoned_ape_tools#48] [Feature Requests]
# What

It would be useful to be able to apply a filter to the UI to enable the user to focus on or modify the specific things they are looking for. I propose the following specific filters:

1. On the stars screen, I'd like to be able to filter by cost of each track. So for example, show me only the stars where science < 100
2. On the stars screen, I'd like to be able to filter by minimum distance to enemy stars. Once done, only stars that are at least that far away will be shown.

3. On the fleets screen, I'd like to be able to filter by fleet size - all fleets size X where low <= X <= high.
4. On the fleets screen, I'd like to be able to filter by all fleets that project a combat in their next hop.
5. On the fleets screen, I'd like to be able to filter by all fleets that will be involved in a losing combat.
6. On the fleets screen, I'd like to be able to filter by fleets that loop versus fleets that don't.

# Why

For the stars filter, I want it to affect upgrade actions. Having applied a filter, if I then do a bulk upgrade, it should only upgrade those stars that pass the filter - the right price and the right distance from the enemy. Hopefully the utility of this is clear.

For the fleets filter, I want mid and endgame to be easier - focusing on only very large fleets will enable you to defend better at the macro level, finding small or inefficiently used fleets will enable optimizing your free fleets, finding looping fleets will enable optimizing the pipeline.

# How

I think the filters can be put right into the UI of the screens in the title bar space above the table. For me it'd be easier to implement a prototype than try to mock it up here.

**Comment by Skilly146** [comment]
This is a really cool idea, especially with the star filters since it would make bulk upgrade actually useful again. 

If you're going to be mocking this up then I think the easiest way to do it would be to directly modify the stars object within universe.js which would mean we don't even need to modify the UI since the client won't even know those stars exist and all the upgrade screens, NPA reports, and other screens will filter themselves.

I think filters might be better implemented with hotkeys plus a main tab in the sidebar (by where NPA Agent is) but we could also just to both since yours would be useful when you look at something and then decide to filter.

Filtering by enemy fleets and stars would also be really useful at the very least to send NPA star reports of only enemy pieces.
